noxfile.py

```diff
@@ -128,6 +129,45 @@ def unit(session):
     default(session)
 
 
+@nox.session(python="3.8")
+def system_emulated(session):
+    import subprocess
+    import signal
+
+    try:
+        subprocess.call(["gcloud", "--version"])
+    except OSError:
+        session.skip("gcloud not found but required for emulator support")
+
+    # Currently, CI/CD doesn't have beta component of gcloud.
+    subprocess.call(
+        ["gcloud", "components", "install", "beta", "cloud-firestore-emulator",]
+    )
+
+    hostport = "localhost:8789"
+    session.env["FIRESTORE_EMULATOR_HOST"] = hostport
+
+    p = subprocess.Popen(
+        [
+            "gcloud",
+            "--quiet",
+            "beta",
+            "emulators",
+            "firestore",
+            "start",
+            "--host-port",
+            hostport,
+        ]
+    )
+
+    try:
+        system(session)
+    finally:
+        # Stop Emulator
+        # os.killpg(os.getpgid(p.pid), signal.SIGKILL)
+        pass
```

Comment:
This one is weird -- it seemed to cause a hang on GH Actions earlier.  Note that it also doesn't *work* locally, as the emulator process daemonizes itself. :( (and there is no `stop` command for it).

Reply:
W00t! b1f0c54